### PR TITLE
Set the GenericObjectPoolConfig type

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

**Comment by sazzad16** [comment]
@gkorland Should we consider it as breaking and mark for major release?

**Comment by gkorland** [comment]
@sazzad16 is it breaking?

**Comment by sazzad16** [comment]
@gkorland AFAIK, no. But my confidence level is a bit low on this.

**Comment by sazzad16** [comment]
@gkorland Please resolve conflicts.

**Comment by gkorland** [comment]
@sazzad16 fixed conflicts

File: src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java [hunk 1]
```diff
@@ -14,31 +14,31 @@
...
       int connectionTimeout, int soTimeout, int infiniteSoTimeout, String user, String password, String clientName) {
     this(nodes, poolConfig, connectionTimeout, soTimeout, infiniteSoTimeout, user, password, clientName, false, null, null, null, null);
   }
 
   /**
    * @deprecated This constructor will be removed in future.
    */
-  @Deprecated
-  public JedisClusterConnectionHandler(Set<HostAndPort> nodes, GenericObjectPoolConfig poolConfig,
+  @Deprecated
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java:40]:
```suggestion
  @Deprecated
```

File: src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java [hunk 2]
```diff
@@ -48,8 +48,8 @@ public JedisClusterConnectionHandler(Set<HostAndPort> nodes, GenericObjectPoolCo
...
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java:51]:
```suggestion
  @Deprecated
```

File: src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java [hunk 3]
```diff
@@ -59,8 +59,8 @@ public JedisClusterConnectionHandler(Set<HostAndPort> nodes, GenericObjectPoolCo
   /**
    * @deprecated This constructor will be removed in future.
    */
-  @Deprecated
-  public JedisClusterConnectionHandler(Set<HostAndPort> nodes, final GenericObjectPoolConfig poolConfig,
+  @Deprecated
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java:62]:
```suggestion
  @Deprecated
```

File: src/main/java/redis/clients/jedis/JedisClusterInfoCache.java [hunk 4]
```diff
@@ -59,15 +59,15 @@ public JedisClusterInfoCache(final GenericObjectPoolConfig poolConfig,
   /**
    * @deprecated This constructor will be removed in future.
    */
-  @Deprecated
-  public JedisClusterInfoCache(final GenericObjectPoolConfig poolConfig, final int connectionTimeout,
+  @Deprecated
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisClusterInfoCache.java:62]:
```suggestion
  @Deprecated
```

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**